### PR TITLE
Fix time intervals to match with proto and time window definition in sum/histogram

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: specification/metrics/datamodel.md [hunk 1]
```diff
@@ -804,12 +804,12 @@ the stream.
 
 - *Cumulative temporality* means that successive data points repeat the starting
   timestamp. For example, from start time T0, cumulative data points cover time
-  ranges (T<sub>0</sub>, T<sub>1</sub>), (T<sub>0</sub>, T<sub>2</sub>),
-  (T<sub>0</sub>, T<sub>3</sub>), and so on.
+  ranges (T<sub>0</sub>, T<sub>1</sub>], (T<sub>0</sub>, T<sub>2</sub>],
```
Comment (pirgeo) [inline specification/metrics/datamodel.md:807]: Would for cumulative values the value of T<sub>0</sub> not be included, as in: 
```suggestion
  ranges [T<sub>0</sub>, T<sub>1</sub>], [T<sub>0</sub>, T<sub>2</sub>],
```
etc.?

File: specification/metrics/datamodel.md [hunk 2]
```diff
@@ -804,12 +804,12 @@ the stream.
 
 - *Cumulative temporality* means that successive data points repeat the starting
   timestamp. For example, from start time T0, cumulative data points cover time
```
Comment (pirgeo) [inline specification/metrics/datamodel.md:806]:
```suggestion
  timestamp. For example, from start time T<sub>0</sub>, cumulative data points cover time
```
For uniformity?

File: specification/metrics/datamodel.md [hunk 3]
```diff
@@ -804,12 +804,12 @@ the stream.
...
-  ranges (T<sub>0</sub>, T<sub>1</sub>), (T<sub>0</sub>, T<sub>2</sub>),
-  (T<sub>0</sub>, T<sub>3</sub>), and so on.
+  ranges (T<sub>0</sub>, T<sub>1</sub>], (T<sub>0</sub>, T<sub>2</sub>],
+  (T<sub>0</sub>, T<sub>3</sub>], and so on.
 - *Delta temporality* means that successive data points advance the starting
   timestamp. For example, from start time T0, delta data points cover time
```
Comment (pirgeo) [inline specification/metrics/datamodel.md:810]:
```suggestion
  timestamp. For example, from start time T<sub>0</sub>, delta data points cover time
```

File: specification/metrics/datamodel.md (hunk 1 above)
Comment (bogdandrutu) [reply specification/metrics/datamodel.md:807]: Based on the proto definition and the specification (other places) it is not.

**Review by jmacd — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]

**Pull request merged**